functions/core-functions.php

```diff
@@ -40,6 +40,39 @@ function FormatMemorySize($Bytes, $Precision = 1) {
+
+   // Preliminary decoding
+   $Str = strip_tags(html_entity_decode($Str, ENT_COMPAT, 'UTF-8'));
+   $Str = strtr($Str, $UrlTranslations);
+   $Str = preg_replace('`[\']`', '', $Str);
+
+   // Test for Unicode PCRE support
+   // On non-UTF8 systems this will result in a blank string.
+   $UnicodeSupport = (preg_replace('`[\pP]`u', '', 'P') != '');
+
+   // Convert punctuation, symbols, and spaces to hyphens
+   if ($UnicodeSupport) {
+      $Str = preg_replace('`[\pP\pS\s]`u', '-', $Str);
+   } else {
+      $Str = preg_replace('`[\s_[^\w\d]]`', '-', $Str);
+   }
+
+   // Lowercase, no trailing or repeat hyphens
+   $Str = preg_replace('`-+`', '-', strtolower($Str));
+   $Str = trim($Str, '-');
+
+   return rawurlencode($Str);
```

Comment:
The rawurlencode() part might not be what we want on the porter side as we usually encode output in Vanilla.